File: android/src/main/java/com/davecoates/rnfirebasebridge/FirebaseBridgeCredentialCache.java
package com.davecoates.rnfirebasebridge;

import com.google.firebase.auth.AuthCredential;

import java.util.HashMap;
import java.util.Map;
import java.util.UUID;

public class FirebaseBridgeCredentialCache {

    static private Map<String, AuthCredential> credentialCache = new HashMap<>();

    static String addCredential(AuthCredential credential) {
        UUID credentialUUID = UUID.randomUUID();
        credentialCache.put(credentialUUID.toString(), credential);
        return credentialUUID.toString();
    }

    static AuthCredential getCredential(String id) {
        return credentialCache.get(id);
    }

}

File: ios/FirebaseBridgeCredentialCache.swift
//
//  FirebaseBridgeCredentialCache.swift
//  move
//
//  Created by Dave Coates on 20/08/2016.
//  Copyright © 2016 Facebook. All rights reserved.
//

import Firebase

class FirebaseBridgeCredentialCache {
  
  static let credentialCache = NSCache<AnyObject, AnyObject>()
  
  static func addCredential(_ credential:FIRAuthCredential) -> String {
    let credentialUUID = UUID.init()
    self.credentialCache.setObject(credential, forKey: credentialUUID.uuidString as AnyObject)
    return credentialUUID.uuidString
  }
  
  static func getCredential(_ id:String) throws -> FIRAuthCredential {
    if let credential = credentialCache.object(forKey: id as AnyObject) as? FIRAuthCredential {
      return credential
    }
    throw FirebaseBridgeError.credentialNotFound()
  }
  
}
